### PR TITLE
Bearer auth documentation

### PR DESCRIPTION
**Pull request by shawn-hurley**
**Describe what this PR does and why we need it**:
Follow on PR for #445 for the documentation
Changes proposed in this pull request
 - Update the auth documentation to document both how to set up auth, and how it is implemented.
 - Update various other docs to include updates.

**Does this PR depend on another PR (Use this to track when PRs should be merged)**
depends-on <PR>
N/A
**Which issue this PR fixes (This will close that issue when PR gets merged)**
N/A

### CODE REVIEW

File: docs/auth.md [hunk 1]
```diff
@@ -146,9 +147,84 @@ the format for the secret changes we will need to create a separate secret for
 just the service catalog today OR we need to add a new `UserServiceAdapter` that
 understands that secret.
 
+### Bearer Auth
+The below section will focus on the bearer token auth.
+
+#### Configuration
+By default, if no authentication is specified the broker will use bearer token auth. The bearer token authentication will use delegated auth from the [kubernetes apiserver](https://github.com/kubernetes/apiserver) library.
+
+The configuration is to grant access, through [kubernetes RBAC](https://kubernetes.io/docs/admin/authorization/rbac/) roles and role-bindings, to the url prefix. The broker has added a configuration option `cluster_url` to specify the url_prefx. This value will default to `ansible-service-broker`.
```
Comment (jmrodri) [inline docs/auth.md]: I'd wrap the line better but the important one is 👍 
TYPO: url_prefx -> url_prefix

File: docs/auth.md [hunk 2]
```diff
@@ -146,9 +147,84 @@ the format for the secret changes we will need to create a separate secret for
...
+  kind: ClusterRole
+  metadata:
+    name: access-asb-role
+  rules:
+  - nonResourceURLs: ["/ansible-service-broker", "/ansible-service-broker/*"]
+    verbs: ["get", "post", "put", "patch", "delete"]
+```
+
+#### Deployment template and secrets
+Here is an example of creating a secret that the service catalog can use, from the above role that will grant access. From the [deployment template](https://github.com/openshift/ansible-service-broker/blob/master/templates/deploy-ansible-service-broker.template.yaml#L224-L248):
```
Comment (jmrodri) [inline docs/auth.md]: I'm having a hard time parsing the first sentence.

Here is an example of creating a secret that the service catalog can use, from the above role that will grant access.

the last part feels confusing: from the above role that will grant access

See if you can reword that a little more clearly. Right now I'm struggling coming up with a better way to phrase it though.

File: docs/auth.md [hunk 3]
```diff
@@ -146,9 +147,84 @@ the format for the secret changes we will need to create a separate secret for
...
+        name: ansibleservicebroker-client
+```
+
 ## Developer section
 
-### Auth design
+### Basic Auth design
 
 The authentication system is built with a set of interfaces to allow for easily
 adding new methods of authentication. The 3 core interfaces are: Provider,
```
Comment (jmrodri) [inline docs/auth.md]: SIGH

**Comment by jmrodri** [comment]
Excellent job failing travis with a document change :) @rthallisey is this the same CI problem we've been seeing lately?

File: docs/auth.md [hunk 4]
```diff
@@ -18,26 +22,23 @@ broker:
...
+The type field specifies the type of authentication to use.
 
 The enabled field allows you to disable a particular auth. This keeps you from
 having to delete the entire section of auth just to disable it.
 
-#### Deployment template
-Typically the broker is configured via a
-[ConfigMap](https://docs.openshift.com/container-platform/3.5/dev_guide/configmaps.html) in a deployment template.
+##### Deployment template
+Typically the broker is configured via a [ConfigMap](https://docs.openshift.com/container-platform/3.5/dev_guide/configmaps.html) in a deployment template.
```
[the remark this answers is not in the document]
Comment (shawn-hurley) [reply docs/auth.md]: I stopped the new line.  It was creating a new line at `via a` I put them on the same line. I thought it looked really odd.

File: docs/auth.md [hunk 5]
```diff
@@ -231,3 +307,66 @@ func (t TrustedUserAuth) GetPrincipal(r *http.Request) (Princpal, error) {
     return TrustedUserPrincipal{username: user.Name}, nil
 }
 ```
+
+### Bearer Auth design
+
+Bearer auth is using the [kubernetes apiserver](https://github.com/kubernetes/apiserver) to do delegated authentication and authorization. Kubernetes team will keep this library up to date.
+
+The first thing that you need to do when setting up the generic api server is to tell it the cert, key, port, and address to listen on. You are going to use the [`SecureServingOptions`](https://github.com/kubernetes/apiserver/blob/master/pkg/server/options/serving.go#L38) to set these values.
```
Comment (djzager) [inline docs/auth.md]: I think you should use a permalink for the kubernetes link.

Comment (djzager) [reply docs/auth.md]: I see a handful of other references to master, I think they should all be permalinks.

Source: have read a doc that pointed to master and it was no longer the line that they had intended in the doc.

**Pull request merged**